modifiers/allocation/modifier.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+from enum import Enum
+from ramble.modkit import *
+
+
+class AllocOpt(Enum):
+    # Experiment resource requests
+    N_RANKS = 1
+    N_NODES = 2
+    N_CORES_PER_TASK = 3
```

Comment:
I think N_CORES_PER_TASK should be N_CORES_PER_NODE

Reply:
I think we need both:

* N_CORES_PER_TASK, this is similar to NUM_OMP_THREADS and refers to the possibility that a task might create its own threads and wants more CPU resources based on this. You can also use both like `N_CORES_PER_TASK=1/NUM_OMP_THREADS=2` to imply that both of the OpenMP threads should be sharing a core
* N_CORES_PER_NODE is I think what you mean to be the system description: how many CPUs are on each node. I listed that as CPUS_PER_NODE below

As a general policy for naming I prefixed *requests* with `N_` and the resources *availability* variables don't have that. Maybe a different prefix should be chosen (although in general I think it's helpful to distinguish using a prefix).

Reply:
I think the renaming of variables (generally using "rank" instead of "task") has made this less confusing (and I added both variables to clarify that each is useful).